modules/transaction-payment/src/lib.rs

```diff
@@ -514,6 +532,58 @@ pub mod module {
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
+		/// Dapp wrap call, and user pay tx fee as provided trading path. this dispatch call should
+		/// make sure the trading path is valid.
+		#[pallet::weight({
+			let dispatch_info = call.get_dispatch_info();
+			(T::WeightInfo::with_fee_path().saturating_add(dispatch_info.weight), dispatch_info.class,)
+		})]
+		pub fn with_fee_path(
+			origin: OriginFor<T>,
+			fee_swap_path: Vec<CurrencyId>,
+			call: Box<CallOf<T>>,
+		) -> DispatchResult {
+			ensure_signed(origin.clone())?;
```

Comment:
should we consider other origin, i.e. root or council account that not issued by user tx?